src/lib/tokenizer.js

```diff
@@ -0,0 +1,87 @@
+class Tokenizer {
+  static theTokenizer;
+
+  constructor(content, literalsList) {
+    this.program = content;
+    this.literals = literalsList;
+    this.tokens = [];
+    this.currentToken = 0;
+
+    this.tokenize();
+  }
+
+  tokenize() {
+    let tokenizedProgram = this.program;
+    tokenizedProgram = tokenizedProgram.replace(/\n/g, '');
+    console.log(this.program);
+    this.literals.forEach(
+        s => {
+          tokenizedProgram = tokenizedProgram.replace(s, `_${s}_`);
+          console.log(tokenizedProgram);
+        });
+    tokenizedProgram = tokenizedProgram.replace(/__/g, '_');
+    console.log(tokenizedProgram);
+    const temparray = tokenizedProgram.split('_');
+    this.tokens = temparray.slice(1, temparray.length);
```

Comment:
Should slice begin at 0? Also, should we trim white spaces and tabs

Reply:
```java
System.arraycopy(temparray,1,tokens,0,temparray.length-1);
System.out.println(Arrays.asList(tokens));
```

According to the Java source code, the tokens are copied starting at 1. So we should slice at    1 as well. See https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/System.html#arraycopy(java.lang.Object,int,java.lang.Object,int,int)